### PR TITLE
Document `mlflow.evaluate()` only use first 10000 rows of large spark dataframe for evaluation

### PR DESCRIPTION
**Pull request by WeichenXu123**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Document mlflow.evaluate() only use first 10000 rows of large spark dataframe for evaluation

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by WeichenXu123** [comment]
Q: 
Do we need to change the behavior `spark_df.limit(10000)` to be `spark_df.sample(...)` ?
Using sample might be more expensive when dataset is large.

File: mlflow/models/evaluation/base.py [hunk 1]
```diff
@@ -738,6 +738,8 @@ def evaluate(
           larger than the configured maximum, these curves are not logged.
 
      - Limitations of evaluation dataset:
+        - If using spark dataframe as evaluation dataset, only the first 10000 rows in the
+          dataframe will be used as evaluation data.
```
Comment (harupy) [inline mlflow/models/evaluation/base.py:742]: I'll add this in the `data` argument description.

```
:param data: if ...
```

**Comment by harupy** [comment]
> Q: Do we need to change the behavior `spark_df.limit(10000)` to be `spark_df.sample(...)` ? Using sample might be more expensive when dataset is large. @dbczumar

@WeichenXu123 What is the motivation for this change?

**Comment by dbczumar** [comment]
> Q: Do we need to change the behavior `spark_df.limit(10000)` to be `spark_df.sample(...)` ? Using sample might be more expensive when dataset is large. @dbczumar

Great question! I think we should keep the `limit()` and not call `sample()` because, with `limit()`, it's easy for users to understand which rows get selected. With `sample()`, it's not clear which rows get selected & may not be reproducible, so users may be confused.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @WeichenXu123 !

**Review by harupy — APPROVED** [review]

**Pull request merged**